internal/database/database.go

```diff
@@ -19,7 +20,33 @@ func (c *Config) SetConnector(connector dialect.Connector) {
 	c.connector = connector
 }
 
-func Connect(config Config, useAdmin bool) (*sql.DB, error) {
+type DB struct {
+	*sql.DB
+	database dialect.Database
+}
+
+// SetDatabase is used for testing purposes
+func (db *DB) SetDatabase(database dialect.Database) {
+	db.database = database
+}
```

Comment:
Is it?

Reply:
yes it should not be used outside of tests

Reply:
Ok, never mind then. My IDE didn't find any callers 😄

Reply:
It's added on a separate branch i want to discuss tomorrow

Reply:
I removed the method because i made the database field public that I don't have to reimplement the methods you mentioned below